src/modules/workspaces/workspaces.usecase.ts

```diff
@@ -1364,6 +1364,149 @@ export class WorkspacesUsecases {
     return this.teamRepository.getTeamUserAndTeamByTeamId(userUuid, teamId);
   }
 
+  async deleteWorkspaceContent(
+    workspaceId: Workspace['id'],
+    user: User,
+  ): Promise<void> {
+    const workspace = await this.workspaceRepository.findById(workspaceId);
+
+    if (!workspace) {
+      throw new NotFoundException('Workspace not found');
+    }
+
+    if (!workspace.isUserOwner(user)) {
+      throw new ForbiddenException('You are not the owner of this workspace');
+    }
+
+    const workspaceUser = await this.userRepository.findByUuid(
+      workspace.workspaceUserId,
+    );
+
+    const rootFolder = await this.folderUseCases.getByUuid(
+      workspace.rootFolderId,
+    );
+
+    await this.folderUseCases.deleteByUser(workspaceUser, [rootFolder]);
+
+    await this.workspaceRepository.deleteById(workspaceId);
+  }
+
+  async transferPersonalItemsToWorkspaceOwner(
+    workspaceId: Workspace['id'],
+    user: User,
+  ): Promise<void> {
+    const workspace = await this.workspaceRepository.findById(workspaceId);
+
+    if (!workspace) {
+      throw new NotFoundException('Workspace not found');
+    }
+
+    if (workspace.isUserOwner(user)) {
+      throw new ForbiddenException('You are the owner of this workspace');
+    }
+
+    const memberWorkspaceUser =
+      await this.workspaceRepository.findWorkspaceUser({
+        workspaceId,
+        memberId: user.uuid,
+      });
+
+    const memberRootFolder = await this.folderUseCases.getByUuid(
+      memberWorkspaceUser.rootFolderId,
+    );
+
+    const foldersInPersonalRootFolder =
+      await this.folderUseCases.getFoldersWithParentInWorkspace(
+        user.uuid,
+        {
+          parentId: memberRootFolder.id,
+          deleted: false,
+          removed: false,
+        },
+        { limit: 1, offset: 0 },
+      );
+
+    const filesInPersonalRootFolder =
+      await this.fileUseCases.getFilesInWorkspace(
+        user.uuid,
+        {
+          folderId: memberRootFolder.id,
+          status: FileStatus.EXISTS,
+          deleted: false,
+          removed: false,
+        },
+        { limit: 1, offset: 0 },
+      );
+
+    if (
+      !foldersInPersonalRootFolder.length &&
+      !filesInPersonalRootFolder.length
+    )
+      return;
+
+    const workspaceNetworkUser = await this.userRepository.findByUuid(
+      workspace.workspaceUserId,
```

Comment:
Hey, when we talk about the owner of the workspace to set as the owner of the files/folders of a user leaving, we are talking about the _user who pays_, not the special user created for the network. The special user will not be used by anyone to list items, thus, if we transfer things to it, they will be invisible to the user who pays. Isn't that the situation?

Reply:
The special network user is used since the moveFolder useCase expects as an argument the user that owns/created both the folder to be moved and the destination folder. In this case it's the network user who owns the folder to be moved (member's personal root folder) and the destination folder (owner's personal root folder).

This call occurs a few lines below and looks like this:
```typescript
const movedFolder = await this.folderUseCases.moveFolder(
      workspaceNetworkUser,
      memberRootFolder.uuid,
      ownerWorkspaceUser.rootFolderId,
    );
```

Reply:
That's true @jzunigax2, good catch, I missed that